package/src/snapshot.rs

```diff
@@ -53,44 +50,51 @@ impl Snapshot {
     /// Snapshots the dependency tree, downloading all necessary git dependencies and finding their exact versions.
     ///
     /// We resolve path dependencies relative to `root_path`.
-    pub fn new(config: Config, root_path: &Path, manifest: &ManifestFile) -> Result<Self, Error> {
+    pub fn new(config: &Config, root_path: &Path, manifest: &ManifestFile) -> Result<Self, Error> {
         Self::new_with_lock(config, root_path, manifest, &LockFile::empty())
     }
 
     pub fn new_with_lock(
-        config: Config,
+        config: &Config,
         root_path: &Path,
         manifest: &ManifestFile,
         lock: &LockFile,
     ) -> Result<Self, Error> {
         let mut ret = Self {
-            config,
             git: HashMap::new(),
             dependency: HashMap::new(),
             manifests: HashMap::new(),
         };
+        let root_pkg = PrecisePkg::Path {
+            path: PathBuf::new(),
+        };
+        ret.manifests.insert(root_pkg.clone(), manifest.clone());
         for (name, dep) in manifest.sorted_dependencies() {
             let lock_entry = lock.dependencies.get(name);
-            ret.snapshot_recursive(root_path, lock, lock_entry, dep, None)?;
+            ret.snapshot_recursive(config, root_path, lock, lock_entry, dep, &root_pkg)?;
         }
         Ok(ret)
     }
 
     // TODO: take in an import sequence (like: the dependency was imported from x, which was imported from y) and use it to improve error messages
     fn snapshot_recursive(
         &mut self,
+        config: &Config,
         root_path: &Path,
         lock: &LockFile,
         lock_entry: Option<&LockFileDep>,
         dep: &Dependency,
-        relative_to: Option<&PrecisePkg>,
+        // TODO: maybe this is simpler if we make this non-optional, and have the root denoted by PrecisePkg with an empty path?
+        // (Or maybe with an absolute path to the root package?)
```

Comment:
Is that comment obsolete? It looks like you've settled down on the root being a `PrecisePkg` with an empty path indeed?

Reply:
Indeed...